src/realm/list.hpp

```diff
@@ -163,85 +215,146 @@ template <>
 void Lst<ObjLink>::do_remove(size_t);
 extern template class Lst<ObjLink>;
 
-class LnkLst : public Lst<ObjKey>, public ObjList {
+// Extern template declarations for lists of primitives:
+extern template class Lst<int64_t>;
+extern template class Lst<bool>;
+extern template class Lst<StringData>;
+extern template class Lst<BinaryData>;
+extern template class Lst<Timestamp>;
+extern template class Lst<float>;
+extern template class Lst<double>;
+extern template class Lst<Decimal128>;
+extern template class Lst<ObjectId>;
+extern template class Lst<UUID>;
+extern template class Lst<util::Optional<int64_t>>;
+extern template class Lst<util::Optional<bool>>;
+extern template class Lst<util::Optional<float>>;
+extern template class Lst<util::Optional<double>>;
+extern template class Lst<util::Optional<ObjectId>>;
+extern template class Lst<util::Optional<UUID>>;
```

Comment:
There are already extern templates for those. We can't get rid of the specializations for `do_insert` etc., because we do some bookkeeping for backlinks and tombstones in those methods. :(